src/dotnet/APIView/APIViewWeb/Repositories/ReviewManager.cs

```diff
@@ -510,6 +520,54 @@ public async void UpdateReviewBackground()
                     _telemetryClient.StopOperation(operation);
                 }
             }
+        }
+
+        public async Task<CodeFile> GetCodeFile(string repoName, string buildId, string artifactName, string packageName, string originalFileName, string codeFileName, MemoryStream memoryStream)
+        {
+            Stream stream = null;
+            CodeFile codeFile = null;
+            if (string.IsNullOrEmpty(codeFileName))
```

Comment:
What cases do we want to only download and extract a single file?

Reply:
If there is a scenario then I believe the download api might take a file pattern you can pass, at least the download task does in DevOps.

Reply:
This is for backward compatibility to support API reviews for PR until we switch over all languages to new sand boxing model. Conditionally allowing either "current" file or new zip allow us to migrate language by language to new model.

Reply:
Oh I missed the fact this isn't extracting a file from a zip it is just downloading a single file.